### PR TITLE
(TK-247) Path Traversal Attack Tests

### PR DESCRIPTION
**Pull request by jpinsonault**
Moved over from a [status-service PR](https://github.com/puppetlabs/trapperkeeper-status/pull/34)

### CODE REVIEW

File: test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_proxy_test.clj [hunk 1]
```diff
@@ -735,4 +735,58 @@
...
+                     ; because the ring handler didn't understand the request, which
+                     ; would have a different message in the body. The request shouldn't
+                     ; get as far as the hello-goodbye-count-ring-handler
+                     (is (re-find #"Problem accessing /hello-proxy" (:body response))))))
+          ; Counter should still be at 0
+          (is (= 0 (deref goodbye-counter))))
+        (testing "counter is working correctly"
+          ; A valid request to bump the counter
+          (http-get "https://localhost:9001/goodbye/" default-options-for-https-client)
+          (is (= 1 (deref goodbye-counter))))))))
```
Comment (fpringvaldsen) [inline test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_proxy_test.clj]: Might be worth making a normal request to `https://localhost:10001/hello-proxy` and `https://localhost:10001/hello-proxy/world` to ensure they return the expected response.

**Comment by KevinCorcoran** [comment]
@jpinsonault @fpringvaldsen @camlow325 If there's any way we could get this merged this morning, that would be :smile_cat:

File: test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_proxy_test.clj (hunk 1 above)
Comment (jpinsonault) [inline test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_proxy_test.clj]: Sounds good

File: test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_proxy_test.clj [hunk 2]
```diff
@@ -735,4 +735,58 @@
...
         (is (= 200 (:status response)))))))
 
-
+; Modified from proxy-ring-handler
+(defn count-ring-handler
+  "Increments counter if the endpoint is /goodbye"
+  [counter req]
+  (condp = (:uri req)
+    "/hello/" {:status 200 :body (str "Hello, World!"
+                                        ((:headers req) "x-fancy-proxy-header")
```
Comment (camlow325) [inline test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_proxy_test.clj]: If nothing validates `x-fancy-proxy-header` or `cookie`, I'd just remove these lines.

**Comment by camlow325** [comment]
I agree with @fpringvaldsen's suggestion on adding an extra assertion to see that the requests to `.../hello-proxy` and `.../hello-proxy/world` work as expected.  :+1: on the rest.

**Comment by jpinsonault** [comment]
@fpringvaldsen Added checks to make sure the proxy was working (good catch, because `/hello/world` was giving a 404)
@camlow325 Got rid of the extra headers

**Comment by fpringvaldsen** [comment]
:+1:

**Comment by KevinCorcoran** [comment]
Going to merge this since it appears to have two :+1:s.  @camlow325 please feel free to follow-up with us if there's anything else on this.

**Comment by camlow325** [comment]
Looks fine to me.

**Pull request merged**